x-pack/plugins/fleet/server/routes/message_signing_service/handlers.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { FleetRequestHandler } from '../../types';
+
+import { defaultFleetErrorHandler } from '../../errors';
+import { appContextService } from '../../services';
+import type { RotateKeyPairSchema } from '../../types/rest_spec/message_signing_service';
+
+export const rotateKeyPairHandler: FleetRequestHandler<
+  undefined,
+  TypeOf<typeof RotateKeyPairSchema.query>,
+  undefined
+> = async (_, request, response) => {
+  if (!request.query?.acknowledge) {
```

Comment:
Interesting concept here. Being that the API is a `POST`, I would suggest moving the required ack to the `body` instead of using query params

Reply:
Also - maybe create a `schema` for it and then you don't have to validate the request specifically in the handler as validation would just fail.

Reply:
I think we cannot have proper custom messages in the schema validator when the field is not added

Reply:
done d895980d1ce97a1fddbbd60817d3ec9cfd89e2a6

Reply:
@dasansol92 let me know if we want the validation to stay in the schema, else I'll revert the schema changes I did.

Reply:
If the custom error message is shown in both cases (when the value is missing and when the value is false) I'm ok having the validation at schema level.

Reply:
let's keep it in the schema then